test/WebSites/DispatchingWebSite/Startup.cs

```diff
@@ -0,0 +1,115 @@
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddDispatcher(options =>
+            {
+                options.DataSources.Add(new DefaultEndpointDataSource(new[]
+                {
+                    new MatcherEndpoint(
+                        (next) => (httpContext) =>
+                        {
+                            var response = httpContext.Response;
+                            var payloadLength = _homePayload.Length;
+                            response.StatusCode = 200;
+                            response.ContentType = "text/plain";
+                            response.ContentLength = payloadLength;
+                            return response.Body.WriteAsync(_homePayload, 0, payloadLength);
+                        },
+                        "/",
+                        new { controller = "Home", action = "Index" },
+                        0,
+                        EndpointMetadataCollection.Empty,
+                        "ActionAsMethod"),
+                    new MatcherEndpoint(
+                        (next) => (httpContext) =>
+                        {
+                            var response = httpContext.Response;
+                            var payloadLength = _homePayload.Length;
+                            response.StatusCode = 200;
+                            response.ContentType = "text/plain";
+                            response.ContentLength = payloadLength;
+                            return response.Body.WriteAsync(_homePayload, 0, payloadLength);
+                        },
+                        "{controller}/{action}",
+                        new { controller = "Home", action = "Index" },
+                        0,
+                        EndpointMetadataCollection.Empty,
+                        "ActionAsMethod"),
+                }));
+            });
+
+            services.AddMvc();
+
+            services.AddScoped<TestResponseGenerator>();
+            services.AddSingleton<IActionContextAccessor, ActionContextAccessor>();
+            services.AddSingleton<EndpointDataSource, MvcEndpointDataSource>();
+        }
+
+        public void Configure(IApplicationBuilder app)
+        {
+            app.UseDispatcher();
+
+            //app.UseMvc(routes =>
+            //{
+            //    routes.MapAreaRoute(
+            //       "flightRoute",
+            //       "adminRoute",
+            //       "{area:exists}/{controller}/{action}",
+            //       new { controller = "Home", action = "Index" },
+            //       new { area = "Travel" });
+
+            //    routes.MapRoute(
+            //        "ActionAsMethod",
+            //        "{controller}/{action}",
+            //        defaults: new { controller = "Home", action = "Index" });
+
+            //    routes.MapRoute(
+            //        "RouteWithOptionalSegment",
+            //        "{controller}/{action}/{path?}");
+            //});
```

Comment:
Assuming this is a placeholder for when we add conventional-routing-via-dispatcher

Reply:
It was just commented out from the copy paste. Conventional-routing-via-dispatcher, I'm not quite clear how it will work. Do you imagine it will use MapRoute and build endpoints from the routes or will it be something else?